src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs

```diff
@@ -53,11 +53,36 @@ public partial class EventCommandEventArgs : System.EventArgs
         public bool DisableEvent(int eventId) { throw null; }
         public bool EnableEvent(int eventId) { throw null; }
     }
-    public partial class EventCounter : System.IDisposable
+    public abstract partial class DiagnosticCounter : System.IDisposable
     {
-        public EventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) { }
+        public DiagnosticCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) { }
+        public void AddMetadata(string key, string value) { }
         public void Dispose() { }
+        public string DisplayName { get { throw null; } set { } }
+        public string Name { get { throw null; } }
+        public System.Diagnostics.Tracing.EventSource EventSource { get { throw null; } }
+    }
+
+    public partial class EventCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public EventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) : base(name, eventSource) { }
         public void WriteMetric(float value) { }
+        public void WriteMetric(double value) { }
+    }
+    public partial class PollingCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public PollingCounter(string name, System.Diagnostics.Tracing.EventSource eventSource, Func<double> metricProvider) : base(name, eventSource) { }
+    }
+    public partial class IncrementingEventCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public IncrementingEventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) : base(name, eventSource) { }
+        public void Increment(double increment = 1) { }
+        public TimeSpan DisplayRateTimeScale { get { throw null; } set { } }
+    }
+    public partial class IncrementingPollingCounter : System.Diagnostics.Tracing.DiagnosticCounter
```

Comment:
Are these the type names agreed to in the API review? Looking at it here, it's strange to me that IncrementingEventCounter didn't derive from EventCounter, and the same for {IncrementingPollingCounter}.

Reply:
Yes, the names were agreed to during the API review meeting. 
`EventCounter` now derives from a base class `DiagnosticCounter`, and all these other counters do as well.